presto-main/src/main/java/com/facebook/presto/Session.java

```diff
@@ -96,6 +100,7 @@ public Session(
         this.remoteUserAddress = requireNonNull(remoteUserAddress, "remoteUserAddress is null");
         this.userAgent = requireNonNull(userAgent, "userAgent is null");
         this.clientInfo = requireNonNull(clientInfo, "clientInfo is null");
+        this.clientTags = requireNonNull(clientTags, "clientTags is null");
```

Comment:
safer to create an immutable copy (similar to what we do with `systemProperties` here).